### PR TITLE
Remove bootstrap style import `@import 'bootstrap/scss/input-group'`

### PR DESCRIPTION
**Pull request by gitstart-sourcegraph**
## Ref
[Parent Issue](https://github.com/sourcegraph/sourcegraph/issues/29200)
[Sourcegraph Issue](https://github.com/sourcegraph/sourcegraph/issues/30281)
[GitStart Ticket](https://app.gitstart.com/clients/sourcegraph/tickets/SG-30281)

## Success Criteria
- [ ] All highlighted imports has been removed
- [ ] All css rules currently used from the imports are now available in style sheets

## Implementation Details
Let's also remove the following imports and copy CSS rules that we need to our CSS files:

1) remove the imports **@import 'bootstrap/scss/input-group** from global `index.scss` file
2) copy styles used in codebase into the global `input-group.scss` file

## ETA
5 - 9 hrs
- Move and modify @Mixins where required according to Chromatic UI changes.
- Move and modify scss variables where required according to Chromatic UI changes.

### CODE REVIEW

**Comment by gitstart-sourcegraph** [comment]
UI Review
https://www.chromatic.com/pullrequest?appId=5f0f381c0e50750022dc6bf7&number=30352&view=changes

**Comment by gitstart-sourcegraph** [comment]
@valerybugakov This is ready for review, kindly take a look.

File: client/branded/src/global-styles/input_group.scss [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+}
+
+:root {
+    --border-radius: 3px;
+    --enable-rounded: true;
+    --input-border-width: 1px;
+    --input-border-radius: 3px;
+    --input-padding-y: 0.375rem;
+    --input-padding-x: 0.75rem;
+    --input-font-size: 1rem; // Assumes the browser default, typically `16px`
```
Comment (valerybugakov) [inline client/branded/src/global-styles/input_group.scss:26]: We have redefined Bootstrap variables here: 

https://github.com/sourcegraph/sourcegraph/blob/25b5aca6ca76336cca48dfa5c9aaa19b25543a8e/client/branded/src/global-styles/index.scss#L10

Changing font-size results in unexpected UI changes [here](https://k8s.sgdev.org/github.com/sourcegraph/sourcegraph-typescript/-/stats/contributors).

File: client/branded/src/global-styles/input_group.scss [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    --font-weight-normal: 400;
+    --input-line-height: 1.5;
```
Comment (valerybugakov) [inline client/branded/src/global-styles/input_group.scss:28]: We already have redefined variable here: 

https://github.com/sourcegraph/sourcegraph/blob/25b5aca6ca76336cca48dfa5c9aaa19b25543a8e/client/branded/src/global-styles/index.scss#L11

Let's make sure that we do not override existing variables.

**Review by valerybugakov — CHANGES_REQUESTED** [review]
Let's compare usages of updated classes in the branch against our dogfood instance or the `main` branch and ensure that we don't have unexpected changes.

**Review by valerybugakov — APPROVED** [review]
⚡️

**Pull request merged**